### PR TITLE
Feature/report types of deploymentin operator

### PR DESCRIPTION
**Pull request by waodim**

### CODE REVIEW

File: deploymentmetadata/deploymentmetadata.go [hunk 1]
```diff
@@ -14,34 +15,106 @@ const (
...
+	}
+
+	if metadata.DynaKube.Spec.InfraMonitoring.Enabled {
+		metadata.osAgentDeployed = "true"
+	}
+
+	if metadata.DynaKube.Spec.ClassicFullStack.Enabled {
+		metadata.osAgentDeployed = "true"
+		metadata.codeModulesDeployed = ""
+	}
```
Comment (meik99) [inline deploymentmetadata/deploymentmetadata.go:115]: should the metrics also report "false" if the features are not set?

File: go.mod [hunk 2]
```diff
@@ -5,15 +5,20 @@ go 1.16
 require (
 	github.com/container-storage-interface/spec v1.3.0
 	github.com/containers/image/v5 v5.9.0
+	github.com/containers/storage v1.23.7 // indirect
+	github.com/daixiang0/gci v0.2.9 // indirect
```
Comment (chrismuellner) [inline go.mod:9]: Did you add this by accident?

Comment (waodim) [reply go.mod:9]: Yes, I will remove it, thank you.

**Comment by DTMad** [comment]
Please rework as discussed

**Comment by DTMad** [comment]
As discussed internally we will adapt the behaviour so we report the usage within each component

**Pull request closed without merging**